### PR TITLE
Add msys2 workflow for CMake

### PR DESCRIPTION
**Pull request by byrnHDF**

### CODE REVIEW

File: .github/workflows/msys2-cmake.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            $GITHUB_WORKSPACE
+
+      - name: CMake Build
+        run: |
+          cmake --build . --parallel 3 --config ${{ inputs.build_mode }}
+        working-directory: ${{ runner.workspace }}/build
+
+      - name: CMake Run Tests
+        run: |
+          ctest . --parallel 2 -C ${{ inputs.build_mode }} -V -E "tfloatsattrs|testhdf5-base|cache_api|dt_arith|dtypes|err_compat"
```
Comment (jhendersonHDF) [inline .github/workflows/msys2-cmake.yml]: Does this action report to cdash? Otherwise, excluding tests just to get this to pass is 100% going to be forgotten about for a long time.

Comment (byrnHDF) [reply .github/workflows/msys2-cmake.yml]: NO but this is not the only one that does this.

Comment (byrnHDF) [reply .github/workflows/msys2-cmake.yml]: Maybe we should have an ENV option that tests all the test to verify they are still failing?

Comment (byrnHDF) [reply .github/workflows/msys2-cmake.yml]: Only daily builds report to CDash

Comment (jhendersonHDF) [reply .github/workflows/msys2-cmake.yml]: It may be more appropriate for msys2 to be a daily build anyway, but I wonder if we can do something with splitting out the failing tests into a separate step with `continue-on-error: true` and report the results somewhere.

Comment (byrnHDF) [reply .github/workflows/msys2-cmake.yml]: I could add a next step that just runs those and add continue on error

Comment (jhendersonHDF) [reply .github/workflows/msys2-cmake.yml]: Right, that's the idea; I'm just wondering if we can have those test results reporting somewhere so that we can keep track of when they start passing/if they're still failing.

Comment (byrnHDF) [reply .github/workflows/msys2-cmake.yml]: The other one is the intel cmake workflow
Those failing tests will show up in the workflow individual - maybe there is a action for email?

Comment (jhendersonHDF) [reply .github/workflows/msys2-cmake.yml]: > Those failing tests will show up in the workflow individual

Sure, I'm just thinking that if the workflow is always passing, no one will think to ever go and check the workflow status again.

File: .github/workflows/nvhpc-cmake.yml [hunk 2]
```diff
@@ -76,6 +76,13 @@ jobs:
           ctest . -E "MPI_TEST|H5TEST-dt_arith|H5TEST-dtransform" --parallel 2 -C ${{ inputs.build_mode }} -V
         working-directory: ${{ runner.workspace }}/build
 
+      - name: CMake Run Expected To Fail Tests
+        shell: bash
+        run: |
+          ctest . -R "H5TEST-dt_arith|H5TEST-dtransform" --parallel 2 -C ${{ inputs.build_mode }} -V
+        working-directory: ${{ runner.workspace }}/build
+        continue-on-error: true
+
```
Comment (derobins) [inline .github/workflows/nvhpc-cmake.yml]: Please put this in a separate PR

**Review by derobins — CHANGES_REQUESTED** [review]
The nvhpc change needs to go in a separate PR

File: .github/workflows/nvhpc-cmake.yml (hunk 2 above)
Comment (byrnHDF) [reply .github/workflows/nvhpc-cmake.yml]: done

**Review by derobins — APPROVED** [review]

**Review by jhendersonHDF — APPROVED** [review]
